src/status_im2/contexts/communities/menus/channel_options/view.cljs

```diff
@@ -0,0 +1,116 @@
+(ns status-im2.contexts.communities.menus.channel-options.view
+  (:require [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [quo2.core :as quo]))
+
+(defn hide-sheet-and-dispatch
+  [event]
+  (rf/dispatch [:bottom-sheet/hide])
```

Comment:
@ibrkhalil, @flexsurfer, @Parveshdhull @status-im/mobile-devs you guys probably noticed this already, but hiding the bottom sheet AND dispatching an event simultaneously can cause a seriously ugly experience because the user sees the bottom sheet being "minimized" and then sees another screen transition.

Screen transitions should almost always be from A to C, not from A to B, and then from B to C. This quickly causes "animation fatigue" and it's not how other major apps work as far as I can tell.

Don't you guys think the ideal solution is to transition directly to the chat, for example? That is, the user should not see 2 screen transitions (bottom sheet + chat navigation).

@ibrkhalil, this is not a suggestion for change. I'm just raising an issue that's plaguing us for some time already.

Reply:
There's an [issue](https://github.com/status-im/status-mobile/issues/15134) for it and it's on my queue

Reply:
> There's an https://github.com/status-im/status-mobile/issues/15134 for it and it's on my queue

Okay nice @ibrkhalil.

But my comment is more about a general problem that dispatching to hide the bottom sheet AND then dispatching an event that causes another screen transition is bad UX. This problem also happens when a notification is pressed on the Activity Center. The issue you're going to work is only going to solve one instance of the problem.

That's why I'd like to hear from other folks if they have a suggestion for a solution and if we're on the same page about this problem. I could open an issue too if other people agree about the general problem.

Reply:
i guess this question is more for the design team, bottom sheet and screen are different transitions, usually they should happen simultaneously, so i would ask design team how ti should work

Reply:
@ilmotta I think when a bottom-sheet it present on the screen, and a button is clicked that will navigate away, the bottom-sheet should be hidden first. A bottom sheet is not a screen. That's how it works on most apps. But of course, it should be done smoothly.

Reply:
just out of curiosity, I have checked a couple of apps (revolut/discord), revolut minimizes and transition to the screen, while with discord is a bit harder to tell what's happening, seems like the two animations happen at the same time (bottom sheet minimizes and screen transition right to left).
in any case ,that's definitely a question for the design team